icepyx/core/spatial.py

```diff
@@ -435,34 +496,56 @@ def __init__(self, spatial_extent, **kwarg):
         if isinstance(spatial_extent, (list, np.ndarray)):
             # bounding box
             if len(spatial_extent) == 4 and all(
-                isinstance(i, scalar_types) for i in spatial_extent
+                isinstance(i, scalar_types)  # pyright: ignore[reportArgumentType]
+                for i in spatial_extent
             ):
                 (
                     self._ext_type,
                     self._spatial_ext,
                     self._geom_file,
-                ) = validate_bounding_box(spatial_extent)
+                ) = validate_bounding_box(
+                    # HACK: Unfortunately, the typechecker can't narrow based on the
+                    # above conditional expressions. Tell the typechecker, "trust us"!
+                    cast(
+                        Union[list[float], NDArray[np.floating]],
```

Comment:
```suggestion
                    cast(
                        Union[list[float, tuple[float,float]], NDArray[np.floating]],
```
I think I've been inconsistent in how I've added tuples to this typing, but didn't want to miss any!

Reply:
Nevermind - missed the len=4 condition!